typing/outcometree.mli

```diff
@@ -25,7 +25,9 @@
 type out_ident =
   | Oide_apply of out_ident * out_ident
   | Oide_dot of out_ident * string
-  | Oide_ident of string
+  | Oide_ident of string ref
+  (** Beware that identifier names can be renamed on the fly to avoid
+      name collision between different identifiers, see {!Printtyp} *)
 
```

Comment:
Wouldn't it be better to define a new type here, rather than use `ref`.
Say, define
```
type out_name = {mutable printed_name: string}
type out_ident =
  ...
  | Oide_ident of out_name
```
My concern is mainly about making the code more robust for maintenance, since you end up adding `ref` in many places, and this could become hard to track.

Reply:
A new record type (or maybe an abstract type) does make the intent clearer here indeed. I have pushed a new commit to this effect.